src/api/goods.ts

```
@@ -1,16 +1,49 @@
 import { Good } from '../types/Good';
 
+/* eslint-disable no-console */
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(`Network response was not ok: ${response.statusText}`);
+      }
+
+      return response.json();
+    })
+    .catch(error => {
+      console.error('Fetch error:', error);
+    });
```

This `.catch` block has a subtle issue. By catching the error here but not re-throwing it, the promise returned by `getAll` resolves with `undefined` instead of being rejected. This violates the function's return type of `Promise<Good[]>` and will cause a runtime error in `App.tsx`. To fix this, you should re-throw the error (e.g., `throw error;`) to ensure the promise is properly rejected, allowing the calling code's `.catch` block to handle it.